### PR TITLE
Fix `libafl_qemu.h`

### LINKED ISSUE #2249: GNU assembler does not recognize `.dword` in `libafl_qemu.h`
**IMPORTANT**
1. You have verified that the issue to be present in the current `main` branch

Thank you for making LibAFL better!

**Describe the bug**
I'm working on a fuzzer under x86-64 using sync exits provided by qemu system mode. I use GCC 11.4.0 to compile the target. It seems that GNU assembler does not recognize the directive `.dword` in the header `libafl_qemu.h`. It can be easily fixed by replacing `.dword` with `.4byte` or `.long`.

**To Reproduce**
1. Include `libafl_qemu.h` in some source files and compile with GCC.

**Expected behavior**
The output is like:
```
test.c: Assembler messages:
test.c:10: Error: unknown pseudo-op: `.dword'
```

### PR DESCRIPTION
**Pull request by rmalmain**
fix #2249.
also added a testcase to avoid this kind of issue in the future

### CODE REVIEW

File: libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs [hunk 1]
```diff
@@ -236,7 +236,8 @@ pub fn merge(
                     .scheduler_mut()
                     .on_remove(&mut state, idx, &Some(testcase))?;
             } else {
-                #[allow(clippy::needless_borrows_for_generic_args)] // False-positive: file_path is used just below
+                #[allow(clippy::needless_borrows_for_generic_args)]
+                // False-positive: file_path is used just below
```
Comment (domenukk) [inline libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs:240]: If anything, the comment should be above the allow

**Comment by domenukk** [comment]
```
Caused by:
  Unable to update https://github.com/nyx-fuzz/libnyx.git?rev=6833d23
```
what's going on?

File: libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs (hunk 1 above)
Comment (rmalmain) [reply libafl_libfuzzer/libafl_libfuzzer_runtime/src/merge.rs:240]: it was autoformated, i'll change this

**Comment by rmalmain** [comment]
> ```
> Caused by:
>   Unable to update https://github.com/nyx-fuzz/libnyx.git?rev=6833d23
> ```
> 
> what's going on?

No idea, it's like this in every run of the CI since today. Maybe a problem on GitHub side? The repo is accessible on my side

**Pull request merged**